### ISSUE TITLE
Gloo Fed - cluster registration fails with k8s v1.24

### ISSUE THREAD

**Issue by bcollard** [solo-io/gloo#7071]
### Gloo Edge Version

1.12.x (latest stable)

### Kubernetes Version

1.24.x

### Describe the bug

Gloo Federation cluster registration works OK until k8s v1.23
It fails with k8s v1.24

### Steps to reproduce the bug

Cluster 1:
- installed Gloo Edge with Gloo-fed enabled by default

Cluster 2: 
- installed Gloo Edge (with the failover Gateway enabled)

Cluster 1:
- ran `glooctl cluster register --cluster-name kind-2-remote  --remote-context kind-2-remote`
- got the following error message:
```
Error: secret for service account not ready yet: All attempts fail:
#1: service account kind-2-remote.gloo-system does not have a token secret associated with it
#2: service account kind-2-remote.gloo-system does not have a token secret associated with it
#3: service account kind-2-remote.gloo-system does not have a token secret associated with it
#4: service account kind-2-remote.gloo-system does not have a token secret associated with it
#5: service account kind-2-remote.gloo-system does not have a token secret associated with it
#6: service account kind-2-remote.gloo-system does not have a token secret associated with it
#7: service account kind-2-remote.gloo-system does not have a token secret associated with it
```

### Expected Behavior

Cluster registration to work.

**Comment by bcollard** [comment]
linked to https://github.com/solo-io/gloo/issues/6611? just a blind guess! :)

**Comment by Charlesthebird** [comment]
This was caused by k8s v1.24, works ok with k8s v1.23.

If running in kind, this will make it use the correct k8s version:
```
kind create cluster --name $CLUSTER --image kindest/node:v1.23.0
```

**Comment by jenshu** [comment]
fix will be available in GlooEE v1.12.41, v1.13.0, and v1.14.0-beta2
